test/e2e/scenarios/remote_dns_name/main_test.go

```diff
@@ -0,0 +1,64 @@
+// Copyright Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the License);
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an AS IS BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build integ
+// +build integ
+
+package remote_dns_name
+
+import (
+	"testing"
+
+	"github.com/openshift-service-mesh/federation/test/e2e"
+	"github.com/openshift-service-mesh/federation/test/e2e/setup"
+
+	"github.com/openshift-service-mesh/federation/test/e2e/setup/coredns"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+)
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite(m).
+		RequireMinClusters(3).
+		RequireMaxClusters(3).
+		Setup(setup.RecreateControlPlaneNamespace).
+		Setup(setup.CreateCACertsSecret).
+		Setup(setup.DeployControlPlanes("k8s")).
+		Setup(coredns.PatchHosts).
+		Setup(setup.InstallOrUpgradeFederationControllers(setup.RemoteAddressDNSName{})).
+		Setup(namespace.Setup(&setup.Namespace, namespace.Config{Prefix: "app", Inject: true})).
+		// a - client
+		// b - service available in east and west clusters - covers importing with WorkloadEntry
+		// c - service available only in west cluster - covers importing with ServiceEntry
+		Setup(setup.Clusters.East.DeployEcho(namespace.Future(&setup.Namespace), "a", setup.WithAllPorts{})).
```

Comment:
If the last argument is a var args, couldn't we skip `setup.WithAllPorts{}`?

Reply:
We could add it as default if empty

Reply:
Or even better - always add it. It replaces part of struct anyway

Reply:
That's reworked now, thanks for the idea!